### PR TITLE
v2 endpoints to list roles and cluster roles

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: v2 endpoints to list roles and cluster roles

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

```release-note
v2 endpoints to list roles and cluster roles:
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/clusterroles
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/clusterrolenames
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/roles
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/rolenames
```

### CODE REVIEW

File: pkg/controller/user-cluster-controller-manager/cluster-role-labeler/cluster_role_labeler.go [hunk 1]
```diff
@@ -23,8 +23,7 @@ import (
 	"go.uber.org/zap"
 
 	predicateutil "k8c.io/kubermatic/v2/pkg/controller/util/predicate"
-	"k8c.io/kubermatic/v2/pkg/handler/v1/cluster"
-
+	handlercommon "k8c.io/kubermatic/v2/pkg/handler/common"
```
Comment (lsviben) [inline pkg/controller/user-cluster-controller-manager/cluster-role-labeler/cluster_role_labeler.go:26]: NIT: put this up with the `k8c` imports

File: pkg/handler/v1/cluster/role_test.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ package cluster_test
 
 import (
 	"fmt"
+	handlercommon "k8c.io/kubermatic/v2/pkg/handler/common"
```
Comment (lsviben) [inline pkg/handler/v1/cluster/role_test.go]: NIT: put it down with the `k8c` imports

**Review by lsviben — COMMENTED** [review]
Looks good, just 2 small comments about import placements

File: pkg/controller/user-cluster-controller-manager/cluster-role-labeler/cluster_role_labeler.go (hunk 1 above)
Comment (zreigz) [reply pkg/controller/user-cluster-controller-manager/cluster-role-labeler/cluster_role_labeler.go:26]: done

File: pkg/handler/v1/cluster/role_test.go (hunk 2 above)
Comment (zreigz) [reply pkg/handler/v1/cluster/role_test.go]: done

File: pkg/handler/common/role.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+	}
+	return apiClusterRole
+}
+
+func convertInternalClusterRoleToExternal(clusterRole *rbacv1.ClusterRole) *apiv1.ClusterRole {
+	return &apiv1.ClusterRole{
+		ObjectMeta: apiv1.ObjectMeta{
+			ID:                clusterRole.Name,
+			Name:              clusterRole.Name,
+			DeletionTimestamp: nil,
```
Comment (moadqassem) [inline pkg/handler/common/role.go]: Nit: will be nil anyway ;)

File: pkg/handler/test/helper.go [hunk 4]
```diff
@@ -30,6 +30,9 @@ import (
 	"testing"
 	"time"
 
+	handlercommon "k8c.io/kubermatic/v2/pkg/handler/common"
```
Comment (moadqassem) [inline pkg/handler/test/helper.go]: Please regroup the imports

**Pull request merged**